### ISSUE TITLE
CheckStyle indentation and tab

### ISSUE THREAD

**Issue by seanangwj** [nus-cs2030/2021-s2#17]
# Summary

Description of the question / problem here.

I faced this issue upon using checkstyle.
<img width="1440" alt="Screenshot 2021-01-30 at 1 06 07 AM" src="https://user-images.githubusercontent.com/78225993/106305587-d2229e80-6297-11eb-8cfa-0748db56b664.png">

## Description:

I do not quite understand what the stated indentation level and expected level means as i do not think that i have 8 or 16 spaces of indentations in my code. Also, i do not know how to really get rid of the tab character even though i have tried removing it.

Clarifications or details with respect to the issue you are facing

## Code Snippets

Attach code snippets (if any)

## Screenshots

Attached is the screenshot

Screenshots (if any)

**Comment by brendancjz** [comment]
Tried helping Sean in this matter but to no avail... Let's see if anyone else face the same issue.

**Comment by Yung-Jer** [comment]
set expandtab
set tabstop=4
set shiftwidth=4
set autoindent
set smartindent
have u included these in your ~/.vimrc?

**Comment by seanangwj** [comment]
Yes, i included a few like set tabstop =4, set shiftwidth =4, set smartindent.

**Comment by seanlowjk** [comment]
The `.vimrc` file might not be the actual one you are using. 

May I know if you are using any IDE to edit your java code?

**Comment by Zekeseoj** [comment]
Hi I am also having this issue, I get my vimrc file using sudo vim .vimrc, then override the file with the above commands. But when i tried to checkstyle it, a ton of errors occurs like OP had. I used ubuntu vim to edit my code. Also to note that my java files are located in my C directory as shown in the 2nd picture. How can i fix this issue?

This shows how i executed my vimrc edit
-
![Screenshot 2021-02-01 182329](https://user-images.githubusercontent.com/77658282/106447195-28702700-64bc-11eb-891f-9306558b0be8.png)

This shows the same/similar issues that OP has.

![Screenshot 2021-02-01 182234](https://user-images.githubusercontent.com/77658282/106447204-2b6b1780-64bc-11eb-9db8-53dbf8367b12.png)

**Comment by nhzaci** [comment]
> Hi I am also having this issue, I get my vimrc file using sudo vim .vimrc, then override the file with the above commands. But when i tried to checkstyle it, a ton of errors occurs like OP had. I used ubuntu vim to edit my code. Also to note that my java files are located in my C directory as shown in the 2nd picture. How can i fix this issue?
> 
> ## This shows how i executed my vimrc edit
> ![Screenshot 2021-02-01 182329](https://user-images.githubusercontent.com/77658282/106447195-28702700-64bc-11eb-891f-9306558b0be8.png)
> 
> This shows the same/similar issues that OP has.
> 
> ![Screenshot 2021-02-01 182234](https://user-images.githubusercontent.com/77658282/106447204-2b6b1780-64bc-11eb-9db8-53dbf8367b12.png)

Hey Zekeseoj,

Could I just check if you have done gg=G after placing those lines inside your .vimrc?

Also, using sudo would not be necessary, just vim will do :)

Please do let us know if you continue to have issues after doing gg=G

**Comment by brendancjz** [comment]
Zeke, input the code Yung Jer sent into your .vimrc file. type gg=G and it should be fine after you compile.

**Comment by Zekeseoj** [comment]
> Zeke, input the code Yung Jer sent into your .vimrc file. type gg=G and it should be fine after you compile.

I did as u said with gg=G and Yung Jer's code but it still continues to give me the errors.
![Screenshot 2021-02-01 190632](https://user-images.githubusercontent.com/77658282/106450800-a33b4100-64c0-11eb-9fb1-3a69177b8b39.png)

**Comment by Zekeseoj** [comment]
Okay I got it, turns out the vimrc file I was editing was not the correct file. Thanks!
